### PR TITLE
Auth end to end

### PR DESCRIPTION
**Pull request by LeshemNoa**
- Removed express-openid-connect from backend: replaced it with jwt verification because the client provides their access token and all the backend has to do is verify it, it doesn't have to manage the authorization flow.
- Implemented optional search result enrichment in the backend for authorized users (adds isSaved per search result based on the user's saved recipes)
- Implemented fetch user recipes - no ui, but action works and brings user recipes to the redux store.
- Implemented starring and unstarring recipes in the ui

### CODE REVIEW

**Comment by hagaishapira** [comment]
This is a really great job!

File: README.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ Create a .env file located at /api/.env and add there:
 - `BASE_URL=`(address sent personally)
 - `APP_SESSION_SECRET=`(info sent personally)
 - `CLIENT_ID=`(info sent personally)
+- `AUTH0_API_AUDIENCE`=https://recipeify.com/users
```
Comment (hagaishapira) [inline README.md:22]: and REACT_APP_AUTH0_AUDIENCE=https://recipeify.com/users in client .env

File: api/server.js [hunk 2]
```diff
@@ -20,12 +22,6 @@ app.use(compression());
 /* vulnerability protection */
 // app.use(helmet());
 
-/* setup auth0 middleware with required authentication for all /api/users routes */
```
[the remark this answers is not in the document]
Comment (LeshemNoa) [reply api/server.js:23]: Take a look at the fixed I just posted https://github.com/recipeify/Recipes/compare/569fb40eae1e5012140a72d22bbcaad4df6dba91..53e7a61560990a3be5c1f4b4f5a2223dc9c7384e and tell me if that's ok

**Comment by hagaishapira** [comment]
@LeshemNoa can't believe you're always force overriding your remote branches at work 😆  how do you work together on a branch?

**Comment by LeshemNoa** [comment]
@hagaishapira it's simple - we don't. Can I merge?

**Comment by hagaishapira** [comment]
@LeshemNoa 
login button is fixed. sidebar isn't.
Plus the redirect somehow doesn't work.

File: README.md (hunk 1 above)
Comment (hagaishapira) [reply README.md:22]: @LeshemNoa

**Pull request merged**